src/OracleService/OracleService.cs

```diff
@@ -0,0 +1,488 @@
+                .Select(u => (u.Contract, u.GetKey()))
+                .ToArray();
+            return accounts != null && accounts.Length > 0;
+        }
+
+        [ConsoleCommand("stop oracle", Category = "Oracle", Description = "Stop oracle service")]
+        private void OnStop()
+        {
+            Interlocked.Exchange(ref CancelSource, null)?.Cancel();
+            if (Timer != null)
+            {
+                Timer.Stop();
+                Timer = null;
+            }
+        }
+
+        public void SendResponseSignature(ulong requestId, byte[] txSign, KeyPair keyPair)
+        {
+            var message = keyPair.PublicKey.ToArray().Concat(BitConverter.GetBytes(requestId)).Concat(txSign).ToArray();
+            var sign = Crypto.Sign(message, keyPair.PrivateKey, keyPair.PublicKey.EncodePoint(false)[1..]);
+            var param = "\"" + Convert.ToBase64String(keyPair.PublicKey.ToArray()) + "\", " + requestId + ", \"" + Convert.ToBase64String(txSign) + "\",\"" + Convert.ToBase64String(sign) + "\"";
+            var content = "{\"id\":" + (++Counter) + ",\"jsonrpc\":\"2.0\",\"method\":\"submitoracleresponse\",\"params\":[" + param + "]}";
+
+            foreach (var node in Nodes)
+            {
+                new Task(() =>
+                {
+                    try
+                    {
+                        var url = new Uri(node);
+                        HttpWebRequest request = (HttpWebRequest)WebRequest.Create(url);
+                        request.Method = "POST";
+                        request.ContentType = "application/json";
+                        using (StreamWriter dataStream = new StreamWriter(request.GetRequestStream()))
+                        {
+                            dataStream.Write(content);
+                            dataStream.Close();
+                        }
+                        HttpWebResponse response = (HttpWebResponse)request.GetResponse();
+                        if (response.ContentLength > ushort.MaxValue) throw new ArgumentOutOfRangeException("The response it's bigger than allowed");
+                        StreamReader reader = new StreamReader(response.GetResponseStream(), Encoding.GetEncoding("UTF-8"));
+                        var retString = reader.ReadToEnd();
+                    }
+                    catch (Exception e)
+                    {
+                        Log($"Failed to send the response signature to {node}, as {e.Message}", LogLevel.Warning);
+                    }
+                }).Start();
+            }
+        }
+
+        public void ProcessRequest(StoreView snapshot, ulong id, OracleRequest req)
+        {
+            Log($"Process oracle request: {req}, txid: {req.OriginalTxid}, url: {req.Url}");
+
+            string data = "";
+            OracleResponseCode code = OracleResponseCode.Success;
+            Uri.TryCreate(req.Url, UriKind.Absolute, out var uri);
+            switch (uri.Scheme.ToLowerInvariant())
```

Comment:
Create a new method:

```c#
OracleResponseCode ProcessUri(Uri uri, out string response)
```

Reply:
I think we shouldn't use `(OracleResponseCode, string)` as the return type. Because if the response code is not `Success`, the `data` is meaningless.

Reply:
Done